fe/src/main/java/org/apache/doris/analysis/BaseViewStmt.java

```diff
@@ -113,27 +113,21 @@ protected void createColumnAndViewDefs(Analyzer analyzer) throws AnalysisExcepti
             return;
         }
 
-        Analyzer tmpAnalyzer = new Analyzer(analyzer);
-        List<String> colNames = cols.stream().map(c -> c.getColName()).collect(Collectors.toList());
-        cloneStmt.substituteSelectList(tmpAnalyzer, colNames);
-        inlineViewDef = cloneStmt.toSql();
-
-        // StringBuilder sb = new StringBuilder();
-        // sb.append("SELECT ");
-        // for (int i = 0; i < columnNames.size(); ++i) {
-        //     if (i != 0) {
-        //         sb.append(", ");
-        //     }
-        //     String colRef = viewDefStmt.getColLabels().get(i);
-        //     if (!colRef.startsWith("`")) {
-        //         colRef = "`" + colRef + "`";
-        //     }
-        //     String colAlias = finalCols.get(i).getName();
-
-        //     sb.append(String.format("`%s`.%s AS `%s`", tableName.getTbl(), colRef, colAlias));
-        // }
-        // sb.append(String.format(" FROM (%s) %s", originalViewDef, tableName.getTbl()));
-        // inlineViewDef = sb.toString();
+        StringBuilder sb = new StringBuilder();
+        sb.append("SELECT ");
+        for (int i = 0; i < finalCols.size(); ++i) {
+            if (i != 0) {
+                sb.append(", ");
+            }
+            String colRef = viewDefStmt.getColLabels().get(i);
+            if (!colRef.startsWith("`")) {
+                colRef = "`" + colRef + "`";
+            }
+            String colAlias = finalCols.get(i).getName();
+            sb.append(String.format("`%s`.%s AS `%s`", tableName.getTbl(), colRef, colAlias));
```

Comment:
Is this work for a function call in select list?

Reply:
It is work. Because the analyze is happen at before. What is doing here is just to wrap another sql outside the previous analyzed sql.